src/models/geograph.py

```diff
@@ -379,31 +391,26 @@ def _load_from_dataframe(
                 new_polygon = polygon.buffer(tolerance)
                 # find the indexes of all polygons which intersect with this one
                 neighbours: List[int] = [
```

Comment:
Related to the above:
Should we factor out this method as `def _find_nearest_neighbors(self, ...)` to make the load function slightly easier to read in case we want to use it elsewhere?

Reply:
I refactored the `.buffer` outside the for loop to make it vectorised so this is now difficult to do unless we ignore the `tolerance` argument. With the new spatial index rtree the code is much more compact anyway.